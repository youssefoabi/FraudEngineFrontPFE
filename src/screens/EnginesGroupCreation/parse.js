import { isNil, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

const parse = enginesGroup => {
  if (isNil(enginesGroup)) return null;

  const { groupId, groupCode, groupName, isEnable, engines } = enginesGroup;

  return {
    groupId,
    groupCode,
    groupName,
    isEnable,
    engines: isNilOrEmpty(engines) ? [] : values(engines),
  };
};
export default parse;
