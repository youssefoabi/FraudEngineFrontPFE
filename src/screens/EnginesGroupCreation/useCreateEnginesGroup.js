import { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { CREATE_ENGINESGROUP_REQUEST } from '../../components/EnginesGroupForm/actions';

export default function useCreateEnginesGroup() {
  const dispatch = useDispatch();

  const createEnginesGroup = useCallback(
    enginesGroup => dispatch({ type: CREATE_ENGINESGROUP_REQUEST, payload: { enginesGroup } }),
    [dispatch],
  );

  return { createEnginesGroup };
}
