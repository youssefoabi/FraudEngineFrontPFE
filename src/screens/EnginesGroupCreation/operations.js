import { call, put, takeLatest } from 'redux-saga/effects';
import { isEmpty, isNil } from 'ramda';

import {
  CREATE_ENGINESGROUP_ERROR,
  CREATE_ENGINESGROUP_REQUEST,
  CREATE_ENGINESGROUP_SUCCESS,
} from '../../components/EnginesGroupForm/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';
import parse from './parse';

export function* createEnginesGroup({ payload }) {
  yield put({ type: PENDING_REQUEST });

  try {
    const { enginesGroup } = payload;
    if (isNil(enginesGroup) || isEmpty(enginesGroup)) return null;

    yield call(clientApi().createEnginesGroup, parse(enginesGroup));

    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: CREATE_ENGINESGROUP_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: CREATE_ENGINESGROUP_ERROR,
      error: { errorMessage: "Erreur d'enregistrement du groupe de moteurs" },
    });
  }

  return true;
}

export default [takeLatest(CREATE_ENGINESGROUP_REQUEST, createEnginesGroup)];
