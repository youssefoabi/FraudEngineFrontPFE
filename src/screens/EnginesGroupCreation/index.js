import React, { useState } from 'react';

import EnginesGroupScreen from '../EnginesGroupScreen';
import useCreateEnginesGroup from './useCreateEnginesGroup';

function EnginesGroupCreation() {
  const [enginesGroup, setEnginesGroup] = useState({ isEnable: false });

  const { createEnginesGroup } = useCreateEnginesGroup();

  return (
    <EnginesGroupScreen
      enginesGroup={enginesGroup}
      action={createEnginesGroup}
      title="CrÃ©ation"
      setEnginesGroup={setEnginesGroup}
    />
  );
}

export default EnginesGroupCreation;
