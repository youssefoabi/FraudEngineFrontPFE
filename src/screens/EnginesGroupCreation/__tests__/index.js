import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import EnginesGroupCreation from '../';

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
}));

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

describe(EnginesGroupCreation.name, () => {
  itRendersCorrectly(<EnginesGroupCreation />);
});
