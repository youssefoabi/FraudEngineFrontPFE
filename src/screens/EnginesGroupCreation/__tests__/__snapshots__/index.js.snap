// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnginesGroupCreation renders correctly 1`] = `
<EnginesGroupScreen
  action={[Function]}
  enginesGroup={
    Object {
      "isEnable": false,
    }
  }
  setEnginesGroup={[Function]}
  title="CrÃ©ation"
/>
`;
